osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/firmware/GetFirmwareVersionsCommandExecutor.java

```diff
@@ -28,12 +28,14 @@
 import org.opensmartgridplatform.dto.valueobjects.smartmetering.ActionRequestDto;
 import org.opensmartgridplatform.dto.valueobjects.smartmetering.ActionResponseDto;
 import org.opensmartgridplatform.dto.valueobjects.smartmetering.FirmwareVersionResponseDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.GetFirmwareVersionQueryDto;
 import org.opensmartgridplatform.dto.valueobjects.smartmetering.GetFirmwareVersionRequestDto;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 @Component
-public class GetFirmwareVersionsCommandExecutor extends AbstractCommandExecutor<Void, List<FirmwareVersionDto>> {
+public class GetFirmwareVersionsCommandExecutor
+        extends AbstractCommandExecutor<GetFirmwareVersionQueryDto, List<FirmwareVersionDto>> {
 
     private static final int CLASS_ID = 1;
     private static final int ATTRIBUTE_ID = 2;
```

Comment:
private static final int CLASS_ID = org.opensmartgridplatform.dlms.interfaceclass.InterfaceClass.DATA.id();
    private static final int ATTRIBUTE_ID = org.opensmartgridplatform.dlms.interfaceclass.attribute.DataAttribute.VALUE.attributeId();

Reply:
existing class, but the boyscout in me wakes up